arm-operationalinsights/2015-03-20/swagger/OperationalInsights.json

```diff
@@ -0,0 +1,1142 @@
+        "operationId": "Search_UpdateSearchResults",
+        "description": "Gets updated search results for a given search query.",
+        "parameters": [
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "Resource Group name of the Log Analytics Workspace"
+          },
+          {
+            "name": "workspaceName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "Log Analytics workspace name"
+          },
+          {
+            "name": "id",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The id of the search that will have results updated."
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK response definition.",
+            "schema": {
+              "$ref": "#/definitions/SearchResultResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": null
+        }
+      }
+    }
+  },
+  "definitions": {
+    "Account": {
+      "properties": {
+        "customerId": {
+          "type": "string",
+          "description": "Gets or sets the unique ID of the account."
+        },
+        "accountName": {
+          "type": "string",
+          "description": "Gets or sets the name of the account."
+        },
+        "workspaceName": {
+          "type": "string",
+          "description": "Gets or sets the DNS valid workspace name associated with the account."
+        },
+        "location": {
+          "type": "string",
+          "description": "Gets or sets the Azure region that the account was created in."
+        }
+      },
+      "description": "Metadata for an Operational Insights account that isn't linked to a workspace."
+    },
+    "Tag": {
+      "properties": {
+        "Name": {
+          "type": "string",
+          "description": "Gets or sets the tag name."
+        },
+        "Value": {
+          "type": "string",
+          "description": "Gets or sets the tag value."
+        }
+      },
+      "description": "A tag of a saved search."
+    },
+    "SavedSearchProperties": {
+      "properties": {
+        "Category": {
+          "type": "string",
+          "description": "Gets or sets the category."
+        },
+        "DisplayName": {
+          "type": "string",
+          "description": "Gets or sets the display name."
+        },
+        "Query": {
+          "type": "string",
+          "description": "Gets or sets the query."
+        },
+        "Version": {
+          "type": "integer",
+          "format": "int64",
+          "description": "Gets or sets the version number."
```

Comment:
Then why do you model it as int64? Wouldn't int32 be more than sufficient? Can this be a string? Can you check in the service code and confirm this? 
We do support string based enums and then you can add an enum constraint `"enum": ["1"]`.

Would like to let you know one more thing. Usually model properties are sent as camelCase or all lowercase on the wire (in JSON format). I see all the properties as PascalCase. This could be a side effect of programmining in C#. Please take a look at the wire format using Fiddler. Looking at the service code can be misleading at times. 
FYI JSON object are case sensitive dictionaries.